app/delivery/send_to_providers.py

```diff
@@ -148,6 +148,10 @@ def update_notification_to_sending(notification, provider):
     dao_update_notification(notification)
 
 
+provider_cache = TTLCache(maxsize=8, ttl=2)
```

Comment:
Am I correct in thinking that maxsize 8 is chosen from 4 choices of `notification_type` multiplied by 2 choices of `international`? And if we were to add another notification type we would need to change this value?

Reply:
2 choices of notification type, because letters don’t use a provider in the same way. So I did 3, rounded up to nearest power of 2, then doubled. The aim was to get a number big enough that we never have to worry about it being a constraint, but not so big we ever have to worry about it consuming a lot of memory.